nntrainer/tensor/weight.cpp

```diff
@@ -40,6 +42,29 @@ void Weight::runVariableInitializer() {
   Tensor &var_ref = getVariableRef();
   const TensorDim dim = var_ref.getDim();
 
+  unsigned int fan_in, fan_out;
+
+  /// @fixme: when unit is equal to one, this does not work we need to rely on
+  /// effective dimension then actual numbers here.
+  switch (dim.rank()) {
+  case 1:
+    fan_out = fan_in = dim.width();
+    break;
+  case 2: /// fully connected layers
+    fan_in = dim.height();
+    fan_out = dim.width();
+    break;
+  case 4: { /// convolution filters, @todo extend this to > 4
```

Comment:
if the dimension of some axis is 1, the rank will be 3 leading to fan_out/fan_in to be 1.

Reply:
Discussed together with below :)